[SN-78] Добавлена генерация ленты новостей

jbkkkkk opened the pull request.

jbkkkkk committed:
[SN-78]
Добавлена генерация ленты новостей

c23p1 commented on src/Infrastructure/Persistence/Repositories/FeedRepository.cs:
```diff
@@ -0,0 +1,35 @@
+        var posts = await _posts.Where(p => personClubsIds.Contains(p.Id)).OrderByDescending(p => p.CreatedAt)
+            .Skip(pagination.PageIndex * pagination.PageSize).Take(pagination.PageSize).ToListAsync();
+        if (posts.Count == 0)
+            posts = await _posts.Where(p => p.ClubId == 9).OrderByDescending(p => p.CreatedAt)
```
Константы следует выносить в поля или переменные для ясности, а также следует в целом изменить Id этого клуба на минимальный по типу 1

c23p1 commented on src/Infrastructure/Persistence/Repositories/FeedRepository.cs:
```diff
@@ -0,0 +1,35 @@
+        var personClubsIds = new List<int>();
+        foreach(var personClub in (await _personsClubsRepository.GetAllByPersonIdAsync(personId)))
+            personClubsIds.Add(personClub.ClubId);
+        var posts = await _posts.Where(p => personClubsIds.Contains(p.Id)).OrderByDescending(p => p.CreatedAt)
```
Мне кажется, на данной строчке критическая ошибка
`_posts.Where(p => personClubsIds.Contains(p.Id))`

c23p1 commented on src/Infrastructure/Persistence/Repositories/FeedRepository.cs:
```diff
@@ -0,0 +1,35 @@
+        _personsClubsRepository = personsClubsRepository;
+    }
+    
+    public async Task<ICollection<PostModel>> GetFeedPostsAsync(int personId, PaginationDTO pagination)
```
Вероятно, оптимальнее будет получать посты каждого клуба, а затем сортировать их, чем проверять у каждого поста его ClubId

c23p1 commented on src/API/Controllers/FeedController.cs:
```diff
@@ -0,0 +1,25 @@
+    
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [HttpGet]
+    public async Task<ActionResult<ICollection<PostDetailDTO>>> GetAll([FromQuery] PaginationDTO pagination)
```
Название метода не показывает, что именно будет им возвращено, в случае с названием "Feed" слово All без указания модели не имеет смысла

jbkkkkk committed:
[SN-78]
Добавлен статический класс для хранения константных Id клубов, GetFeedAsync изменено GetFeedPostsAsync, исправлена ошибка с выборкой постов по Id клуба.

c23p1 commented on src/Infrastructure/Constants/ClubIds.cs:
```diff
@@ -0,0 +1,6 @@
+
+public static class ClubIds
+{
+    public const int MireaClubId = 1;
```
Перенести константу в сервис

jbkkkkk committed:
[SN-78]
Внесены незначительные изменения
